src/utils/Dom.ts

```diff
@@ -129,7 +129,25 @@ export class Dom {
    */
   public empty(): void {
     while (this.el.firstChild) {
-      this.el.removeChild(this.el.firstChild);
+      this.removeChild(this.el.firstChild);
+    }
+  }
+
+  /**
+   * Safely removes a child, only suppressing `NotFoundError` if the child's parent was changed.
+   */
+  public removeChild(child: Node) {
+    const oldParent = child.parentNode;
+    try {
+      this.el.removeChild(child);
+    } catch (e) {
+      if ((e as Error).name === 'NotFoundError') {
+        if (oldParent === child.parentNode) {
+          throw e;
+        }
+      } else {
+        throw e;
+      }
```

Comment:
I think the above sequence would be easier to reason if we inverted the condition. e.g.

```
catch (e) {
    if ((e as Error).name !== 'NotFoundError') {
        throw e;
    }

    if (oldParent === child.parentNode) {
        throw e;
    }
}

Reply:
Good idea, I'll simplify it like such.